### PR TITLE
Add avro source generation doc

### PR DESCRIPTION
**Pull request by AdrianRaFo**
## What this does?
Adds a section to the docs about how the Avro IDL is generated to Scala source.

Some whitespaces and blank lines on the existing doc and move it to the index section about this.

Required for the LW workshop

# Checklist

- [x] Reviewed the diff to look for typos, println and format errors.
- [x] Updated the docs accordingly.

### CODE REVIEW

File: docs/src/main/tut/generate-sources-from-idl/avro.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
+---
+layout: docs
+title: Generating sources from Avro
+permalink: /generate-sources-from-avro
+---
+
+# Generating sources from Avro
+
+In this section we are going to explain how we can generate the different Scala structures using the `Avro` IDL.
```
Comment (fedefernandez) [inline docs/src/main/tut/generate-sources-from-idl/avro.md]: I'd add another mention here for `avrohugger`

File: docs/src/main/tut/generate-sources-from-idl/avro.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+The name of that protocol will be the name of our Scala file.
+
+```avroidl 
+protocol People { 
+ ... 
+} 
+```
+
+***srcGen =>***
```
Comment (juanpedromoreno) [inline docs/src/main/tut/generate-sources-from-idl/avro.md]: Maybe it worth mentioning that `srcGen` is an sbt command that is run (by default) in the `compile` phase

File: docs/src/main/tut/generate-sources-from-idl/avro.md [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+  
+```scala
+final case class PeopleResponse(result: Either[Errors.Value, Person])
+```
+
+### Coproducts
+
+And finally, when we have **`three or more non-null types`** on a single `union`, 
+we'll have a [shapeless](https://github.com/milessabin/shapeless/wiki/Feature-overview:-shapeless-2.0.0)' `Coproduct` on the same order as well:
```
Comment (juanpedromoreno) [inline docs/src/main/tut/generate-sources-from-idl/avro.md]:
```suggestion
we'll have a [shapeless](https://github.com/milessabin/shapeless/wiki/Feature-overview:-shapeless-2.0.0#coproducts-and-discriminated-unions)' `Coproduct` on the same order as well:
```

File: docs/src/main/tut/generate-sources-from-idl/avro.md [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+layout: docs
+title: Generating sources from Avro
+permalink: /generate-sources-from-avro
+---
+
+# Generating sources from Avro
+
+In this section we are going to explain how we can generate the different Scala structures using the `Avro` IDL.
+
+## Protocols
```
Comment (juanpedromoreno) [inline docs/src/main/tut/generate-sources-from-idl/avro.md]:
```suggestion
## Avro Protocols
```

File: docs/src/main/tut/generate-sources-from-idl/avro.md [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+
+```avroidl
+record PeopleResponse {
+  union{ string, int, Errors } result;
+}
+```
+
+***srcGen =>***
+
+```scala
```
Comment (juanpedromoreno) [inline docs/src/main/tut/generate-sources-from-idl/avro.md]: Could we make these blocks as `tut` blocks so we will be forced to keep up them up-to-date?

File: docs/src/main/tut/generate-sources-from-idl/avro.md [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+  import idl "People.avdl"; //Under the same folder
+
+  example.protocol.PeopleResponse getPerson(example.protocol.PeopleRequest request);
+
+}
+```
+
+***srcGen =>***
+
+```scala
```
Comment (juanpedromoreno) [inline docs/src/main/tut/generate-sources-from-idl/avro.md]: Same here, using `tut` would be great

File: docs/src/main/tut/generate-sources-from-idl/index.md [hunk 7]
```diff
@@ -118,66 +125,67 @@ The following example shows how to set up a dependency with another artifact or
...
-      def SayHelloProto(req: HelloRequest): F[HelloResponse]
-      def LotsOfRepliesProto(req: HelloRequest): Stream[F, HelloResponse]
-      def LotsOfGreetingsProto(req: Stream[F, HelloRequest]): F[HelloResponse]
-      def BidiHelloProto(req: Stream[F, HelloRequest]): Stream[F, HelloResponse]
-      def BidiHelloFs2Proto(req: Stream[F, HelloRequest]): Stream[F, HelloResponse]
-    }
-    
-    }
- ```
+```scala
```
Comment (juanpedromoreno) [inline docs/src/main/tut/generate-sources-from-idl/index.md]: tut?

File: docs/src/main/tut/generate-sources-from-idl/avro.md [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+  val NotFound, Duplicated, None = Value
+}
+```
+
+## Unions
+
+`Unions` are a complex `Avro` type for fields inside `record`s. 
+As its name suggest, it represents a type composed by another types.
+
+Depending on the types composing the `union`, `Mu` will interpret it on different ways:
```
Comment (mrtmmr) [inline docs/src/main/tut/generate-sources-from-idl/avro.md]: Should we use `mu` or `Mu`?  We should have the same format in every section

Comment (AdrianRaFo) [reply docs/src/main/tut/generate-sources-from-idl/avro.md]: Good question, it's a proper name so I think it should be capitalized but we usually use the lowecased version. @juanpedromoreno @fedefernandez thoughts?

Comment (juanpedromoreno) [reply docs/src/main/tut/generate-sources-from-idl/avro.md]: I don't have a strong preference here, I guess that the capitalized version is the more commonly used: `Mu`

Comment (fedefernandez) [reply docs/src/main/tut/generate-sources-from-idl/avro.md]: I usually use `mu` but the capitalized one is ok for me ¯\_(ツ)_/¯

**Pull request merged**